calyx/src/ir/structure.rs

```diff
@@ -39,6 +41,23 @@ impl Port {
     pub fn is_hole(&self) -> bool {
         matches!(&self.parent, PortParent::Group(_))
     }
+
+    /// Gets name of parent object.
+    pub fn get_parent_name<'a>(&'a self) -> Id {
+        match &self.parent {
+            PortParent::Cell(cell) => {
+                cell.upgrade().unwrap().borrow().name.clone()
+            }
+            PortParent::Group(group) => {
+                group.upgrade().unwrap().borrow().name.clone()
+            }
+        }
+    }
+
+    /// Gets a key usable in a hashmap from a port
+    pub fn key(&self) -> (Id, Id) {
```

Comment:
This seems far too specific to the HashMap used in the inliner.

Reply:
I can move this for now, but I thought it might be useful in general for using Ports in hashmaps. This seems potentially useful for other anlayses.